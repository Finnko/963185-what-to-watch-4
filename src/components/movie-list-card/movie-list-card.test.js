import React from 'react';
import renderer from 'react-test-renderer';
import {createHistory} from 'history';
import {Router} from 'react-router-dom';
import MovieListCard from './movie-list-card';

const mock = {
  id: `1`,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  thumb: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
};

describe(`MovieListCard component render correctly`, () => {
  it(`Should MovieListCard component render correctly`, () => {
    const history = createHistory();

    const tree = renderer
      .create(
          <Router history={history}>
            <MovieListCard
              movie={mock}
            />
          </Router>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
