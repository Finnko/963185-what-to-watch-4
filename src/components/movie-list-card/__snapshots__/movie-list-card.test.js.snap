// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListCard component render correctly Should MovieListCard component render correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="small-movie-card__link"
    href="/movie/1"
    onClick={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="100%"
      />
    </div>
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/movie/1"
      onClick={[Function]}
    >
      Fantastic Beasts: The Crimes of Grindelwald
    </a>
  </h3>
</article>
`;
